### PR TITLE
[14.0][ADD] documents_stock_production_lot

### PR DESCRIPTION
**Pull request by Vimal-OSI**
Task: https://pm.opensourceintegrators.com/web#id=20980&model=project.task&view_type=form&menu_id=176

### CODE REVIEW

File: documents_stock_production_lot/models/documents_workflow_rule.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2021 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, fields, models
```
Comment (bodedra) [inline documents_stock_production_lot/models/documents_workflow_rule.py]:
```suggestion
from odoo import fields, models, _
```

File: documents_stock_production_lot/models/documents_workflow_rule.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+from odoo.exceptions import UserError
+
+
+class DocumentsWorkflowRule(models.Model):
+    _inherit = ["documents.workflow.rule"]
+
+    def _compute_business(self):
+        return self._get_business()
+
```
Comment (bodedra) [inline documents_stock_production_lot/models/documents_workflow_rule.py:11]:
```suggestion

```

File: documents_stock_production_lot/models/documents_workflow_rule.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    has_business_option = fields.Boolean(default=True, compute="_compute_business")
```
Comment (bodedra) [inline documents_stock_production_lot/models/documents_workflow_rule.py:12]:
```suggestion
    has_business_option = fields.Boolean(default=True, compute="_get_business")
```

File: documents_stock_production_lot/models/documents_workflow_rule.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    create_model = fields.Selection(
+        selection_add=[("stock.production.lot", "Lot/Tracking Number")]
+    )
+
+    def create_record(self, documents=None):
+        res = super().create_record(documents=documents)
+        if self.create_model == "stock.production.lot":
+            lot = self.env.company.lot_id
```
Comment (bodedra) [inline documents_stock_production_lot/models/documents_workflow_rule.py]: may be company from user? :thinking: 

> self.env.user.company_id.lot_id

**Review by bodedra — CHANGES_REQUESTED** [review]

File: documents_stock_production_lot/data/product.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<odoo noupdate="1">
+
+    <record id="documents_product" model="product.product">
+        <field name="lot_folder" ref="documents_folder_lot" />
+    </record>
+
+</odoo>
```
Comment (max3903) [inline documents_stock_production_lot/data/product.xml:7]: @Vimal-OSI What is the purpose of this record?

File: documents_stock_production_lot/models/res_company.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        domain=_domain_company,
+        default=lambda self: self.env.ref(
+            "documents_stock_production_lot.documents_folder_lot",
+            raise_if_not_found=False,
+        ),
+    )
+    document_lot_tags = fields.Many2many(
+        "documents.tag", "document_lot_tags", string="Lot/Tracking Number Tags"
+    )
+    lot_id = fields.Many2one("stock.production.lot", "Existing Lot/Tracking Number")
```
Comment (max3903) [inline documents_stock_production_lot/models/res_company.py:26]: @Vimal-OSI What is the purpose of this field?

File: documents_stock_production_lot/models/res_config_settings.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        readonly=False,
+        string="Lot Tags",
+    )
+    lot_id = fields.Many2one(
+        "stock.production.lot",
+        related="company_id.lot_id",
+        readonly=False,
+        string="Existing Lot/Tracking Number",
+        config_parameter="lot_id",
+    )
```
Comment (max3903) [inline documents_stock_production_lot/models/res_config_settings.py:33]: @Vimal-OSI What is the purpose of this field?

File: documents_stock_production_lot/models/documents_workflow_rule.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+                    attachment_copy = document.attachment_id.with_context(
+                        no_document=True
+                    ).copy()
+                    this_document = document.copy({"attachment_id": attachment_copy.id})
+                this_document.write(
+                    {
+                        "res_model": lot._name,
+                        "res_id": lot.id,
+                    }
+                )
```
Comment (max3903) [inline documents_stock_production_lot/models/documents_workflow_rule.py]: @Vimal-OSI This is linking the documents to the same lot... Not sure this is the expected behavior. We don't want to create a new lot, we want to attach the document to an existing lot.

File: documents_stock_production_lot/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    "license": "LGPL-3",
+    "depends": ["documents", "stock"],
+    "data": [
+        "data/documents_folder.xml",
+        "data/documents_facet.xml",
+        "data/res_company.xml",
+        "data/documents_workflow_rule.xml",
+        "views/res_config_settings.xml",
+    ],
+    "maintainers": ["bodedra"],
```
Comment (bodedra) [inline documents_stock_production_lot/__manifest__.py:22]:
```suggestion
    "maintainers": ["max3903"],
```

File: documents_stock_production_lot/models/documents_workflow_rule.py (hunk 3 above)
Comment (max3903) [reply documents_stock_production_lot/models/documents_workflow_rule.py:12]: @bodedra Compute methods should start with `_compute`

Comment (bodedra) [reply documents_stock_production_lot/models/documents_workflow_rule.py:12]: @max3903 that's true. I added suggestion based on Odoo Enterprise implementation on the documents modules. ref: documents_****

**Comment by Vimal-OSI** [comment]
All the suggested changes are done.
Thank you.

**Review by bodedra — APPROVED** [review]
Code review only :+1:

**Review by max3903 — APPROVED** [review]

**Pull request merged**